### PR TITLE
Clean up usages of Random in chests

### PR DESCRIPTION
**Pull request by BenCheung0422**
When populating chests, as it involves world generation, it should also include the world generation random provider when using randoms. This fixes this problem. Unused `Inventory#tryAdd` methods are also removed. All chests should be the same for the same seed in initialization now.

### CODE REVIEW

**Comment by Litorom** [comment]
This looks like it fixes another bug I discovered today, the dungeon chests in structures generate empty. Please fix your code and test and I’ll merge it

File: src/client/java/minicraft/saveload/Load.java [hunk 1]
```diff
@@ -1129,7 +1129,7 @@ private static Entity getEntity(String string, int mobLevel) {
 			case "Workbench": return new Crafter(Crafter.Type.Workbench);
 			case "Chest": return new Chest();
 			case "DeathChest": return new DeathChest();
-			case "DungeonChest": return new DungeonChest(false);
+			case "DungeonChest": return new DungeonChest(null);
```
Comment (Makkkkus) [inline src/client/java/minicraft/saveload/Load.java:1132]: These will probably cause nullpointerexceptions.

Comment (BenCheung0422) [reply src/client/java/minicraft/saveload/Load.java:1132]: `null` case is handled within `DungeonChest`.

Comment (Makkkkus) [reply src/client/java/minicraft/saveload/Load.java:1132]: Alright, I see what you mean. But do we want the chests to be empty in those cases. And should the populate inventory function even be called from the DungeonChest class?

Comment (BenCheung0422) [reply src/client/java/minicraft/saveload/Load.java:1132]: In `Load`, this method fetches the initial state (empty state this case) for further entity loading, although actually this is quite badly designed.

Comment (Litorom) [reply src/client/java/minicraft/saveload/Load.java:1132]: Will empty locked chest populate?

Comment (BenCheung0422) [reply src/client/java/minicraft/saveload/Load.java:1132]: If it is loaded from saves, the chests are never populated.

Comment (BenCheung0422) [reply src/client/java/minicraft/saveload/Load.java:1132]: Also, the initial state of "locked" does not affect the final state of the chest. The saved value of "locked" should always be loaded from the save after the initial construction of the instance.

Comment (Litorom) [reply src/client/java/minicraft/saveload/Load.java:1132]: Locked dungeon chests that spawn in structures method are empty while those that are spawned via the other method are populated.

Comment (BenCheung0422) [reply src/client/java/minicraft/saveload/Load.java:1132]: image

Comment (BenCheung0422) [reply src/client/java/minicraft/saveload/Load.java:1132]: I hope I have not misunderstood.

Comment (Litorom) [reply src/client/java/minicraft/saveload/Load.java:1132]: I’ll do more testing, just fix the conflicts too please.

**Pull request merged**